third_party/amd/python/triton_amd.cc

```diff
@@ -345,4 +410,100 @@ void init_triton_amd(py::module &&m) {
   m.def("add_scalarize_packed_fops_llvm_pass", [](llvm::Function *fn) {
     mlir::triton::AMD::runScalarizePackedFOpsPass(*fn);
   });
+
+  auto hipBlas = m.def_submodule("hipblas");
+  py::class_<HipBlasLtInstance>(hipBlas, "HipBlasLt")
+      .def(py::init<>([&](py::object &workspace) {
+        auto wrk_ptr = workspace.attr("data_ptr")().cast<uint64_t>();
+        auto wrk_size = workspace.attr("numel")().cast<size_t>() *
+                        workspace.attr("element_size")().cast<size_t>();
+        return new HipBlasLtInstance(wrk_ptr, wrk_size);
+      }))
+      .def("matmul",
+           [](HipBlasLtInstance &self, py::object &A, py::object &B,
+              py::object &C) {
+             auto A_ptr = A.attr("data_ptr")().cast<uint64_t>();
+             auto B_ptr = B.attr("data_ptr")().cast<uint64_t>();
+             auto C_ptr = C.attr("data_ptr")().cast<uint64_t>();
+
+             auto A_shape = A.attr("shape").cast<std::vector<int>>();
+             auto B_shape = B.attr("shape").cast<std::vector<int>>();
+             auto C_shape = C.attr("shape").cast<std::vector<int>>();
+
+             auto A_dtype =
+                 A.attr("dtype").attr("__str__")().cast<std::string>();
+             auto B_dtype =
+                 B.attr("dtype").attr("__str__")().cast<std::string>();
+             auto C_dtype =
+                 C.attr("dtype").attr("__str__")().cast<std::string>();
+
+             checkMatmulConstraints(A_dtype, B_dtype, C_dtype, A_shape, B_shape,
+                                    C_shape);
+
+             std::string dtype_str =
+                 A_dtype.substr(A_dtype.find_last_of('.') + 1);
+             hipDataType dtype;
+
+             if (dtype_str == "float8_e4m3fnuz") {
+               dtype = HIP_R_8F_E4M3_FNUZ;
+             } else if (dtype_str == "float16") {
+               dtype = HIP_R_16F;
+             } else if (dtype_str == "float32") {
+               // Use FP32 inputs with TF32 compute in hipblasLt (set in compute
+               // type)
+               dtype = HIP_R_32F;
+             } else if (dtype_str == "bfloat16") {
+               dtype = HIP_R_16BF;
+             } else {
+               throw std::runtime_error(
+                   "Unsupported dtype for hipblasLt.matmul: " + dtype_str);
+             }
+             self.matmul(A_shape[0], B_shape[0], A_shape[1], A_ptr, B_ptr,
+                         C_ptr, dtype);
+           })
+      .def("gemm", [](HipBlasLtInstance &self, py::object &A, py::object &B,
```

Comment:
This is almost the same as the above--can we create some utility function to dedup?